pytensor/tensor/rewriting/subtensor.py

```diff
@@ -1576,7 +1577,15 @@ def local_blockwise_advanced_inc_subtensor(fgraph, node):
         x = alloc(x, *batch_shape, *core_shape)
 
     new_idxs = [slice(None)] * batch_ndim + new_idxs
-    symbolic_idxs = x[tuple(new_idxs)].owner.inputs[1:]
+    x_view = x[tuple(new_idxs)]
+
+    # We need to introduce any implicit expand_dims on core dimension of y
+    y_core_ndim = y.type.ndim - batch_ndim
+    if (missing_y_core_ndim := x_view.type.ndim - batch_ndim - y_core_ndim) > 0:
+        missing_axes = tuple(range(batch_ndim, batch_ndim + missing_y_core_ndim))
+        y = expand_dims(y, missing_axes)
+
+    symbolic_idxs = x_view.owner.inputs[1:]
```

Comment:
nit: i'd put this back up under where `x_view` is declared. There's nothing happening between lines 1583-1586 that implicate it, but putting it here suggests that there is.

Reply:
The lines in between use xview, namely to check how many dims it has to know whether y is being implicitly bpradcasted

Reply:
oh it's the stupid walrus